packages/lit-dev-content/site/tutorials/content/first-component/06.md

```diff
@@ -0,0 +1,139 @@
+In this section you'll make it so when you click on the `word-viewer`, the
+animation plays in reverse.
+
+New internal state is needed to represent the current animation `playDirection`.
+
+{% switchable-sample %}
+
+```ts
+// word-viewer.ts
+
+  @state() private playDirection = 1;
+
+```
+
+```js
+// word-viewer.js
+
+  static properties = {
+    playDirection: {state: true},
+    ...
+  };
+
+  constructor() {
+    ...
+    this.playDirection = 1;
+  }
+
+```
+
+{% endswitchable-sample %}
+
+Next, update the math that finds the displayed `word` in the `render()` method
+to account for a negative `this.idx`.
+
+{% switchable-sample %}
+
+```ts
+// word-viewer.ts
+
+  render() {
+    ...
+    const word = splitWords[((this.idx % splitWords.length) + splitWords.length) % splitWords.length];
+    ...
+  }
+
+```
+
+```js
+// word-viewer.js
+
+  render() {
+    ...
+    const word = splitWords[((this.idx % splitWords.length) + splitWords.length) % splitWords.length];
+    ...
+  }
+
+```
+
+{% endswitchable-sample %}
+
+Change `tickToNextWord` to choose the next word based on the `playDirection`.
+
+{% switchable-sample %}
+
+```ts
+// word-viewer.ts
+
+  tickToNextWord = () => { this.idx += this.playDirection; };
+
+```
+
+```js
+// word-viewer.js
+
+  tickToNextWord = () => { this.idx += this.playDirection; };
+
+```
+
+{% endswitchable-sample %}
+
+And add a `switchPlayDirection` method to invert the  `playDirection`.
+
+
+{% switchable-sample %}
+
+```ts
+// word-viewer.ts
+
+  switchPlayDirection() {
+    this.playDirection *= -1;
+  }
+
+```
+
+```js
+// word-viewer.js
+
+  switchPlayDirection() {
+    this.playDirection *= -1;
+  }
+
+```
+
+{% endswitchable-sample %}
+
+Finally, bind `this.switchPlayDirection` to invoke when the `<pre>` element is
+clicked using Lit's [declarative event listener
+syntax](/docs/components/events/#adding-event-listeners-in-the-element-template).
+
+{% switchable-sample %}
+
+```ts
+// word-viewer.ts
+
+  render() {
+    ...
+    return html`<pre
+      @click=${this.switchPlayDirection}
```

Comment:
You do link to the doc but is it worth pointing out the handler passed in here does not need binding unlike `tickToNextWord`?

https://lit.dev/docs/components/events/#understanding-this-in-event-listeners

Reply:
Done!